opentelemetry/proto/profiles/v1development/eventprofile.md

```diff
@@ -0,0 +1,284 @@
+        clock_id:          0
+    } 
+    { # index 1
+        name_string_index: 2  # "thread_state"
+        clock_id:          0
+    }
+]
+clocks: [
+    { # index 0
+        frequency :      1_000_000_000          # 1 Ghz clock, i.e. 1 cycle = 1 ns
+        time_unix_nanos: 1_257_894_000_000_000_000
+    }
+]
+attributes: [
+    { # index 0
+        key_index: 4          # thread.id
+        value: {int_value: 1} # 1
+    },
+    { # index 1
+        key_index: 4,         # thread.id
+        value: {int_value: 2} # 2
+    }
+    { # index 2
+        key_index: 5,         # thread.state
+        value: {key_index: 6} # mutex
+    }, # index 3
+    {
+        key_index: 5,         # thread.state
+        value: {key_index: 7} # sleep
+    }
+]
+strings: [
+    "",             # index 0
+    "cpu_sample",   # index 1
+    "thread_state", # index 2
+    "ms",           # index 3
+    "thread.id",    # index 4
+    "thread.state", # index 5
+    "mutex",        # index 6
+    "sleep",        # index 7
+]
+
+events: [
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 1          # main;serve_request
+        time:        10_500_000 # 10.5ms since clock start
+        attributes:  [0]        # thread_id: 1
+    }
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 2          # main;foo
+        time:        300_000    # 0.3ms since previous event
+        attributes: [1]         # thread_id: 2
+    },
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 1          # main;serve_request
+        time:        9_200_000  # 9.2ms since previous event
+        attributes:  [0]        # thread_id: 1
+    },
+    {
+        event_type_index: 1            # thread_state
+        stack_index:      1            # main;serve_request;mutex_lock
+        time:             300_000_000  # 300ms since previous event
+        duration:         50_000_500   # 50ms
+        attributes:       [0, 2]       # thread_id: 1, thread.state: mutex
+    },
+    {
+        event_type_index: 1             # thread_state
+        stack_index:      1             # main;foo;sleep
+        time:             5_000_000     # 5ms since previous event
+        duration:         1_000_000_500 # 1s
+        attributes:       [1, 3]        # thread_id: 2, thread.state: sleep
+    },
+]
+# stack, location, function, mapping, etc. omitted for brevity
+```
+### Aggregated Memory Profile
+
+Below is an example of an aggregated memory profile like those available in Go.
+
+```txtpb
+event_types: [
+    { # index 0
+        name_string_index: 1 # allocations
+        unit_string_index: 3 # By (bytes)
+    }
+    { # index 1
+        name_string_index: 2 # liveheap
+        unit_string_index: 3 # By (bytes)
+    }
+]
+strings: [
+    "",              # index 0
+    "allocations",   # index 1
+    "liveheap",      # index 2
```

Comment:
I think it's necessary to differentiate between virtual memory, physical memory, and situations like page faults. Typically, virtual memory consumed is more than physical memory because some memory regions might be zeroed and never referenced.  Some might be 'copy-on-write'.

Reply:
This needs to be clarified, probably via semantic conventions. But the example given here is allocations profiled by the Go runtime. Those are always measured in terms of virtual memory. I wrote up more details on this [here](https://www.datadoghq.com/blog/go-memory-metrics/).